javapoet/src/main/java/com/palantir/javapoet/TypeSpec.java

```diff
@@ -459,6 +475,28 @@ void emit(CodeWriter codeWriter, String enumName, Set<Modifier> implicitModifier
         }
     }
 
+    /**
+     * Emits the constructor parameters for a record type. i.e. {@code (String name, int count)}.
+     */
+    private void emitRecordConstructor(CodeWriter codeWriter) throws IOException {
```

Comment:
This isn't a "constructor" it's a "record header"

https://docs.oracle.com/javase/specs/jls/se17/html/jls-8.html#jls-8.10

```suggestion
    private void emitRecordHeader(CodeWriter codeWriter) throws IOException {
```